rs/messaging/src/routing/stream_handler.rs

```diff
@@ -991,6 +1005,67 @@ impl StreamHandlerImpl {
         }
     }
 
+    /// Credits the cycles attached to a refund message to its respective receiver.
+    ///
+    /// Returns the amount of attached cycles that were lost iff the recipient was
+    /// deleted (i.e. should have been hosted by this subnet but does not exist).
+    fn induct_refund(
+        &self,
+        refund: &Refund,
+        state: &mut ReplicatedState,
+        stream: &mut Stream,
+    ) -> Cycles {
+        // Subnet that should have received the message according to the routing table.
+        let receiver_host_subnet = state
+            .metadata
+            .network_topology
+            .route(refund.recipient().get());
+
+        match receiver_host_subnet {
+            // Matching receiver subnet, try crediting the cycles.
+            Some(host_subnet) if host_subnet == self.subnet_id => {
```

Comment:
```suggestion
            Some(self.subnet_id) => {
```
personal preference.

Reply:
I also thought of that at some point. Unfortunately Rust syntax does not allow that. It has to be a pattern.

Reply:
Probably the same reason you can't match into `Arcs`. In that case nevermind.